packages/scm/src/browser/scm-widget.tsx

```diff
@@ -154,540 +133,45 @@ export class ScmWidget extends ReactWidget implements StatefulWidget {
-        const icon = labelProvider.getIcon(resource.sourceUri);
-        const color = decorations && decorations.color || '';
-        const letter = decorations && decorations.letter || '';
-        const tooltip = decorations && decorations.tooltip || '';
-        const relativePath = new URI(rootUri).relative(resource.sourceUri.parent);
-        const path = relativePath ? relativePath.toString() : labelProvider.getLongName(resource.sourceUri.parent);
-        return <div key={String(resource.sourceUri)}
-            className={`scmItem ${ScmWidget.Styles.NO_SELECT}${repository.selectedResource === resource ? ' ' + SELECTED_CLASS : ''}`}
-            onContextMenu={this.renderContextMenu}
-            onMouseEnter={this.showHover}
-            onMouseLeave={this.hideHover}
-            ref={this.detectHover}
-            onClick={this.handleClick}
-            onDoubleClick={this.handleDoubleClick} >
-            <span className={icon + ' file-icon'} />
-            <div className='noWrapInfo' >
-                <span className='name'>{name}</span>
-                <span className='path'>{path}</span>
-            </div>
-            <ScmInlineActions {...{
-                hover,
-                menu: menus.getMenu(ScmWidget.RESOURCE_INLINE_MENU),
-                args: this.contextMenuArgs,
-                commands,
-                contextKeys,
-                group: resource.group
-            }}>
-                <div title={tooltip} className='status' style={{ color }}>
-                    {letter}
-                </div>
-            </ScmInlineActions>
-        </div >;
-    }
-
-    protected open = () => this.props.resource.open();
-
-    protected selectChange = () => this.props.repository.selectedResource = this.props.resource;
-
-    protected readonly contextMenuPath = ScmWidget.RESOURCE_CONTEXT_MENU;
-    protected get contextMenuArgs(): any[] {
-        return [this.props.resource];  // TODO support multiselection
-    }
-
-    /**
-     * Handle the single clicking of nodes present in the widget.
-     */
-    protected handleClick = () => {
-        // Determine the behavior based on the preference value.
-        const isSingle = this.props.corePreferences && this.props.corePreferences['workbench.list.openMode'] === 'singleClick';
-        this.selectChange();
-        if (isSingle) {
-            this.open();
-        }
-    };
+    export namespace Factory {
 
-    /**
-     * Handle the double clicking of nodes present in the widget.
-     */
-    protected handleDoubleClick = () => {
-        // Determine the behavior based on the preference value.
-        const isDouble = this.props.corePreferences && this.props.corePreferences['workbench.list.openMode'] === 'doubleClick';
-        // Nodes should only be opened through double clicking if the correct preference is set.
-        if (isDouble) {
-            this.open();
+        export interface WidgetOptions {
+            readonly order?: number;
```

Comment:
Are these properties used or just copied?

Reply:
These should not be here.  As you say, just copied and I failed to remove them.

Reply:
These should not be here.  As you say, just copied and I failed to remove them.